src/docs/sphinx/quickstart.rst

```diff
@@ -267,28 +267,268 @@ must be specified using either:
+By making ``mpic++`` point to ``mpicxx``, this command will also automatically point to the correct executable if you change the global MPI package through MacPorts in the future.
+
+Python environment
+^^^^^^^^^^^^^^^^^^
+
+The Serac build scripts install the ``clingo`` package in your Python environment (and may even *uninstall* it if it finds it with a version it considers too old). 
+I don't like to let software packages change my Python environment, particularly by adding or removing packages with ``pip``, as this can lead to version dependency conflicts.  
+I create a Python virtual environment in the serac root directory with the built-in ``venv`` module. 
+
+.. (I use the name "venv" for the virtual environment; We might consider adding `venv/` to the list of git ignores in the root directory).
+
+.. code-block:: bash
+
+   $ python3 -m venv <virtual-env-name>
+
+Since spack will need ``clingo``, activate the python vitual environment and 
+install the package to meet this dependency:
+
+.. code-block:: bash
+
+   $ source <virtual-env-name>/bin/activate
+   $ pip install clingo
+
+The python virtual environment must be active when invoking the ``uberenv.py`` script below.
+
+Configuring spack
+^^^^^^^^^^^^^^^^^
+
+In ``compilers.yaml``:
+
+.. code-block:: yaml
+
+   - compiler:
+      environment: {}
+      extra_rpaths: []
+      flags: {}
+      modules: []
+      operating_system: bigsur
+      paths:
+         cc: /opt/local/bin/clang
+         cxx: /opt/local/bin/clang++
+         f77: /opt/local/bin/gfortran
+         fc: /opt/local/bin/gfortran
+      spec: clang@12.0.1
+      target: x86_64
+
+NOTES: 
+
+* The `operating_system` field should be set to ``bigsur`` for macOS 11. Refer to the spack documentation for other versions of macOS.
+* By default, MacPorts installs packages in ``/opt/local``; the above paths need to be adjusted if you choose a different location.
+  This of course applies to the packages in ``packages.yaml`` as well.
+* As noted above, the ``port select ...`` commands will set which version of clang gets invoked by the executables ``/opt/local/bin/clang``, etc.
+  The paths above are thus valid only if you activated the ``clang`` package that matches the compiler spec.
+  Alternatively, you could set the full name and path of the executables of the desired compilers if you don't want MacPorts settings to interfere with your spack settings.
+* You should set the spec to be equal to the actual version of the compiler you installed.
+
+In ``packages.yaml``:
+
+.. code-block:: yaml
+
+   packages:
+   all:
+      compiler: [clang, gcc]
+      providers:
+         blas: [netlib-lapack]
+         lapack: [netlib-lapack]
+         mpi: [openmpi]
+
+   mpi:
+      buildable: false
+   openmpi:
+      externals:
+      - spec: openmpi@4.1.4
```

Comment:
In theory, you should be able to use `spack external find openmpi`, but you may have to add flags, like `-p /opt/local` and `--scope` with whatever scope you're putting the `packages.yaml` in -- `uberenv`'s configuration might dictate what the right scope is (e.g., `system`, `user`). See https://spack.readthedocs.io/en/latest/command_index.html#spack-external-find for details.